### PR TITLE
Mul scalar

### PR DESCRIPTION
**Pull request by 4rmanc**

### CODE REVIEW

**Comment by lazovicff** [comment]
Few things left to do:
- #161 
- Add SelectTableChipset

**Comment by 4rmanc** [comment]
@lazovicff [This](https://github.com/eigen-trust/protocol/blob/mul_scalar/circuit/src/ecc/mod.rs#L670) variable gives a warning. Should I put `_` before it to counter this warning? What can be the optimal way to solve this warning?

File: circuit/src/integer/rns.rs [hunk 1]
```diff
@@ -313,6 +309,25 @@ impl RnsParams<Fq, Fr, 4, 68> for Bn256_4_68 {
...
 
+	fn construct_mul_qr(a_bn: BigUint, b_bn: BigUint) -> ([Fr; 4], [Fr; 4]) {
+		let wrong_mod_bn = Self::wrong_modulus();
+		let (quotient, result_bn) = (a_bn * b_bn).div_rem(&wrong_mod_bn);
+		let q = decompose_big::<Fr, 4, 68>(quotient);
+		let result = decompose_big::<Fr, 4, 68>(result_bn);
+		(q, result)
+	}
+
+	fn construct_div_qr(a_bn: BigUint, b: Integer<Fq, Fr, 4, 68, Self>) -> ([Fr; 4], [Fr; 4]) {
```
Comment (lazovicff) [inline circuit/src/integer/rns.rs:320]: Can we have a consistent interface here?
We should pass `a_bn: BigUint, b: BigUint, b_invert: BigUint`.
And also add a check `assert!(b * b_invert == 1)`

File: circuit/src/edwards/mod.rs [hunk 2]
```diff
@@ -13,6 +13,17 @@ use halo2::{
...
+pub fn assigned_as_bool<F: FieldExt>(bit: AssignedCell<F, F>) -> bool {
+	let bit_value = bit.value();
+	let mut is_one = false;
+	bit_value.map(|f| {
+		is_one = F::one() == *f;
+		f
+	});
+	is_one
+}
+
```
Comment (lazovicff) [inline circuit/src/edwards/mod.rs:26]: Can we move this to `utils.rs`

**Review by duguorong009 — APPROVED** [review]

**Review by lazovicff — APPROVED** [review]

**Pull request merged**